### PR TITLE
Change rust.yml to run benchmark

### LINKED ISSUE #7707: Benchmark CI job doesn't actually run benchmark
### Describe the bug

In `rust.yml`, `verify-benchmark-results` seems intended to run benchmark like as follows:
```
cargo test serde_q --package datafusion-benchmarks --profile release-nonlto --features=ci -- --test-threads=1
```

But there are no tests which includes `serde_q` in the name.
Actually, e752a9cd5390db718d6512d843a changed the name of the benchmark tests.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sarutak**
## Which issue does this PR close?

Closes #7707 

## Rationale for this change
Run benchmarks  as intended.

## What changes are included in this PR?
This PR changes `rust.yml` to run benchmark.
This PR also changes the cargo command running benchmark to narrow the package to be built, which can reduce the build time.

## Are these changes tested?
Done by CI.

## Are there any user-facing changes?
No.

### CODE REVIEW

**Comment by sarutak** [comment]
Now the benchmark actually runs.
https://github.com/apache/arrow-datafusion/actions/runs/6366576587/job/17284455771#step:6:735

And build time is significantly shorter (Total duration: 26m 38s).
https://github.com/apache/arrow-datafusion/actions/runs/6366576587

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -226,17 +226,6 @@ jobs:
...
-        uses: actions/cache@v3
-        with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            ./target/
-          # this key equals the ones on `linux-build-lib` for re-use
-          key: cargo-cache-benchmark-${{ hashFiles('datafusion/**/Cargo.toml', 'benchmarks/Cargo.toml') }}
```
Comment (viirya) [inline .github/workflows/rust.yml:239]: Why remove this? Related to the issue?

Comment (sarutak) [reply .github/workflows/rust.yml:239]: Oh, [this change](https://github.com/apache/arrow-datafusion/pull/7706) is mixed.

**Review by viirya — APPROVED** [review]

**Review by alamb — APPROVED** [review]
👏 

Screenshot 2023-10-01 at 6 34 03 AM

Very nice -- thank you @sarutak 

Also, I would like to thank you again for your efforts to make the DataFusion CI work better and more efficiently. It is most appreciated. 

I actually spent some more time looking into what `plan_q` does and it is not at all clear to me it needs to be built in release mode either

https://github.com/apache/arrow-datafusion/blob/46cdb8c2dc495e8063a0adc5c3f9ac82b136e72e/benchmarks/src/tpch/run.rs#L297-L453

It seems like it just verifies the plans (rather than actually running them). I'll see what happens if I change to using debug (rather than release) mode which compiles faster.

**Pull request merged**